### PR TITLE
Bratseth/user flavors

### PR DESCRIPTION
**Pull request by bratseth**
@freva please review

### CODE REVIEW

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/responses/node5.json [hunk 1]
```diff
@@ -6,14 +6,13 @@
   "hostname": "host5.yahoo.com",
   "parentHostname": "dockerhost2.yahoo.com",
   "openStackId": "node5",
-  "flavor": "docker",
-  "canonicalFlavor": "docker",
+  "flavor": "[vcpu: 1.0, memory: 1.0 Gb, disk 100.0 Gb]",
```
Comment (freva) [inline node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/responses/node5.json]: This is used (among others) as dimension for metrics, so may break some graphs @yngveaasheim

**Review by freva — COMMENTED** [review]
A few places where the type is now `NodeResources` or refers to resources in some way, but the variable name is still `flavor`, not sure if you wanna fix those.

There's also a test failure in `node-admin` due to `node4` has changed CPU 0.2 -> 1.0

File: config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/NodesSpecification.java [hunk 2]
```diff
@@ -171,13 +170,65 @@ public static NodesSpecification dedicated(int count, ConfigModelContext context
...
+        }
+    }
+
+    private static double parseGbAmount(String byteAmount) {
+        byteAmount = byteAmount.strip();
+        byteAmount = byteAmount.toUpperCase();
+        if (byteAmount.endsWith("B"))
+            byteAmount = byteAmount.substring(0, byteAmount.length() -1);
+
+        double multiplier = 1/1000^3;
```
[the remark this answers is not in the document]
Comment (bratseth) [reply config-model/src/main/java/com/yahoo/vespa/model/builder/xml/dom/NodesSpecification.java]: Ffs ... thanks!

File: node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/responses/node5.json (hunk 1 above)
Comment (bratseth) [reply node-repository/src/test/java/com/yahoo/vespa/hosted/provision/restapi/v2/responses/node5.json]: If we have graphs where *node* (not host) flavor is a dimension we'll have to change them when this goes into use.

**Pull request merged**